src/tests/test_double_compare.cpp

```diff
@@ -0,0 +1,220 @@
+}
+
+TEST_F(DoubleCompare, AddNumbers) {
+    std::unique_ptr<std::byte[]> seven_seven(double_type.MakeValue(7.7));
+    std::unique_ptr<std::byte[]> eight_eight(double_type.MakeValue(8.8));
+    std::unique_ptr<std::byte[]> expect(double_type.MakeValue(16.5));
+
+    double_type_ref.Add(seven_seven.get(), eight_eight.get(), result);
+    ASSERT_EQ(double_type_ref.Compare(result, expect.get()), model::CompareResult::kEqual);
+
+    seven_seven.reset(double_type.MakeValue(777.777));
+    eight_eight.reset(double_type.MakeValue(888.888));
+    expect.reset(double_type.MakeValue(1666.665));
+    double_type_ref.Add(seven_seven.get(), eight_eight.get(), result);
+
+    ASSERT_EQ(double_type_ref.Compare(result, expect.get()), model::CompareResult::kEqual);
+}
+
+TEST_F(DoubleCompare, SubNumbers) {
+    std::unique_ptr<std::byte[]> seven_seven(double_type.MakeValue(7.7));
+    std::unique_ptr<std::byte[]> eight_eight(double_type.MakeValue(8.8));
+    std::unique_ptr<std::byte[]> expect(double_type.MakeValue(7.7 - 8.8));
+
+    double_type_ref.Sub(seven_seven.get(), eight_eight.get(), result);
+    ASSERT_EQ(double_type_ref.Compare(result, expect.get()), model::CompareResult::kEqual);
+
+    seven_seven.reset(double_type.MakeValue(777.777));
+    eight_eight.reset(double_type.MakeValue(888.888));
+    expect.reset(double_type.MakeValue(-111.111));
+    double_type_ref.Sub(seven_seven.get(), eight_eight.get(), result);
+    ASSERT_EQ(double_type_ref.Compare(result, expect.get()), model::CompareResult::kEqual);
+}
+
+TEST_F(DoubleCompare, MulNumbers) {
+    std::unique_ptr<std::byte[]> seven_seven(double_type.MakeValue(7.7));
+    std::unique_ptr<std::byte[]> eight_eight(double_type.MakeValue(8.8));
+    std::unique_ptr<std::byte[]> expect(double_type.MakeValue(7.7 * 8.8));
+
+    double_type_ref.Mul(seven_seven.get(), eight_eight.get(), result);
+    ASSERT_EQ(double_type_ref.Compare(result, expect.get()), model::CompareResult::kEqual);
+
+    seven_seven.reset(double_type.MakeValue(777.777));
+    eight_eight.reset(double_type.MakeValue(888.888));
+    expect.reset(double_type.MakeValue(691356.641976));
+    double_type_ref.Mul(seven_seven.get(), eight_eight.get(), result);
+    ASSERT_EQ(double_type_ref.Compare(result, expect.get()), model::CompareResult::kEqual);
+}
+
+TEST_F(DoubleCompare, DivNumbers) {
+    std::unique_ptr<std::byte[]> many_eight(double_type.MakeValue(8888.8888));
+    std::unique_ptr<std::byte[]> few_eight(double_type.MakeValue(8.8));
+    std::unique_ptr<std::byte[]> expect(double_type.MakeValue(8888.8888 / 8.8));
+
+    double_type_ref.Div(many_eight.get(), few_eight.get(), result);
+    ASSERT_EQ(double_type_ref.Compare(result, expect.get()), model::CompareResult::kEqual);
+
+    expect.reset(double_type.MakeValue(8.8 / 8888.8888));
+    double_type_ref.Div(few_eight.get(), many_eight.get(), result);
+    ASSERT_EQ(double_type_ref.Compare(result, expect.get()), model::CompareResult::kEqual);
+}
```

Comment:
All these methods have the same logic, it should be moved out to a function and reused here